packages/notifi-react/lib/context/NotifiTargetContext.tsx

```diff
@@ -661,6 +664,156 @@ export const NotifiTargetContextProvider: FC<PropsWithChildren> = ({
+  //       isCBW: true,
+  //     },
+  //   });
+  //   // await signCoinbaseSignature(address, senderAddress);
+  //   await frontendClient.verifyCbwTarget({
+  //     input: {
+  //       targetId: targetId,
+  //     },
+  //   });
+  // };
+
+  const signCoinbaseSignature = useCallback(
+    async (
+      address: string,
+      senderAddress: string,
+      conversationTopic: string,
+    ) => {
+      const nonce = await createCoinbaseNonce();
+      if (!nonce) throw Error('Unable to sign the wallet. Please try again.');
+
+      const message = `Coinbase Wallet Messaging subscribe\nAddress: ${address}\nPartner Address: ${senderAddress}\nNonce: ${nonce}`;
+
+      const signature = await getSignature(message);
+
+      if (!signature)
+        throw Error('Unable to sign the wallet. Please try again.');
+
+      const payload = {
+        address,
+        nonce,
+        signature: signature as `0x${string}`,
+        isActivatedViaCb: true,
+        partnerAddress: senderAddress,
+        conversationTopic,
+      };
+
+      return await subscribeCoinbaseMessaging(payload);
+    },
+    [getSignature],
+  );
+
+  const xmtpXip42Impl = useCallback(async () => {
+    try {
+      const options: any = {
+        persistConversations: false,
+        env: 'production',
+      };
+      const address = walletWithSignParams.walletPublicKey;
+
+      const signer = {
+        getAddress: (): Promise<string> => {
+          return new Promise((resolve) => {
+            resolve(address);
+          });
+        },
+        signMessage: async (
+          message: ArrayLike<number> | string,
+        ): Promise<string> => {
+          return getSignature(message);
+        },
+      };
+
+      const client = await xmtp.initialize({ options, signer });
+
+      if (client === undefined) {
+        throw Error('XMTP client is uninitialized. Please try again.');
+      }
+
+      // TODO: get senderAddress from target
+      const senderAddress = '0xE80E42B5308d5b137FC137302d571B56907c3003';
+      const conversation = await client.conversations.newConversation(
+        senderAddress,
+      );
+      const conversationTopic = conversation.topic.split('/')[3];
+
+      await client.contacts.allow([senderAddress]);
+
+      await signCoinbaseSignature(address, senderAddress, conversationTopic);
+
+      const targetId = targetData?.wallet?.data?.id ?? '';
+
+      await frontendClient.verifyXmtpTargetViaXip42({
+        input: {
+          web3TargetId: targetId,
+          accountId: address,
+          conversationTopic,
+        },
+      });
+
+      await frontendClient
+        .fetchData()
+        .then(refreshTargetDocument)
+        .catch((e: unknown) => console.error(e));
+
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }, [
+    walletWithSignParams,
+    targetData,
+    signCoinbaseSignature,
+    frontendClient,
+    xmtp,
+    getSignature,
+  ]);
+
+  const signWallet = useCallback(async () => {
```

Comment:
Just being curious, do we have to have the extra layer for this `signWallet`?

Reply:
yes, I think so, if we need additional step